opm/utility/ECLGraph.cpp

```diff
@@ -274,6 +279,19 @@ namespace {
                 std::vector<T>
                 scatterToGlobal(const std::vector<T>& x) const;
 
+                /// Map input vector to active grid cells.
+                ///
+                /// \param[in] x Input vector, defined on the explicitly
+                ///              active cells, all global cells or some
+                ///              other subset (e.g., all non-neighbouring
+                ///              connections).
+                ///
+                /// \return Input vector mapped to active cells or unchanged
```

Comment:
> The caller must ultimately be aware of which use case it is invoking

The caller must be aware of what the end result must be (a vector on active cells). 

> I do not really see the problem?

The problem is that apart from the pore-volume vector, which must always be defined on all global cells, we do not know how big any particular vector will be until we input the values from the result set.  Therefore, in order to support your suggestion, we must equip the public interface of `CartesianGridData::CartesianCells` with a way classifying `x.size()` as

  * All global cells
  * Explicitly active cells (`ACTNUM != 0`)
  * Other

whereas the current strategy enables the caller to say: "I don't care how you do it, just restrict this vector to your active cells."